### ISSUE TITLE
Stop making blocking calls to solr to render collection members...

### ISSUE THREAD

**Issue by jkeck** [sul-dlss/SearchWorks#1989]
...(in search results only?)

**Comment by saseestone** [comment]
@jkeck has a design question before this is fully actionable

**Comment by jkeck** [comment]
@jvine a couple of questions.

1. Do we need to do anything wrt accessibility (aria attributes?) if we are to load collection members asynchronously via JavaScript.
1. What should we do in the case of a non-js browser?  (Maybe some sort of link to all the collection members w/ some special text).  This would be content that would effectively be asynchronously replaced/enhanced w/ the collection member list/filmstrip.
1. We can either asynchronously load all record's collection members on page load, or we could also do something based on what's in the visible viewport (or even just below the viewport so we can finish loading the collection members before the user has scrolled to the particular record).  This may be more of an implementation detail, but wasn't sure if you would have any opinion on this or not.

**Comment by cbeer** [comment]
Is there additional information we could index for collections to make this easier?

**Comment by jvine** [comment]
1. Technically, we would need a placeholder div so the layout of the page would not change when the collection members are added, and would use `aria-live="polite"` to let screen readers know when the placeholder is replaced by content. (In looking at this, I discovered a larger accessibility issue with the filmstrip, which I'll document separately.)
2. There is already a link to the collection members that is separate from the preview section. We will likely be re-wording that link as part of the (likely) SDR / Online presentation chunk of work. Maybe also suppress the placeholder div mentioned in # 1, so we're not holding space for content that won't be loaded.  
3. I would say that loading the viewport-visible content as it's needed would be fine, except that a screen reader can access all the content in the DOM without changing the visible viewport. If it were just lazy-loading images, that would be irrelevant to the SR, but with actual content... I don't know. 

That said, I'm having a really hard time predicting how async loading of multiple portions of the page without user action would actually be presented by a screen reader, and not finding useful examples or parallels.

**Comment by jkeck** [comment]
Really quick analysis on a rough local implementation looks like a possible reduction of the load time of a typical digital collections result (e.g. the 1st page of 20 collections) by about ~45%.

**Comment by jkeck** [comment]
One complicating factor/question is that we can't know if we're going to build a file list or an filmstrip until we've already made the request to get the collection members (the type of display we get is a function of the formats of the collection members returned).  Since the filmstrip and the file list have different heights, adding a placeholder element w/ the correct hight is not knowable until we've already made the blocking request.

Is there a placeholder element hight that we can use w/o knowing if it will be a filmstrip or a file list that replaced the placeholder?
